### PR TITLE
Docs/more dns info

### PR DESCRIPTION
**Pull request by etiennnr**
**How to categorize this PR?**
/area documentation
/kind enhancement

**What this PR does / why we need it**:
Enhancing DNS documentation to add new features and easier to read content

**Release note**:
```other operator
doc user
```

### CODE REVIEW

File: docs/usage/dns_names.md [hunk 1]
```diff
@@ -25,95 +25,121 @@ spec:
...
+## Gardener DNS extension
+The managed external DNS records feature of the Gardener clusters makes all this easier. You do not need DNS service provider specific knowledge, and in fact you do not need to leave your cluster at all to achieve that. You simply annotate the Ingress / Service that needs its DNS records managed and it will be automatically created / managed by Gardener.
+
+Managed external DNS records are supported with the following DNS provider types:
+- aws-route53
+- azure-dns
+- azure-private-dns
+- google-clouddns
+- openstack-designate
+- alicloud-dns
```
Comment (MartinWeindel) [inline docs/usage/dns_names.md:53]:
```suggestion
- alicloud-dns
- cloudflare-dns
```

File: docs/usage/dns_names.md [hunk 2]
```diff
@@ -25,95 +25,121 @@ spec:
...
+
+### Request DNS records for Ingress resources
+
+To request a DNS name for an Ingress or Service object in the shoot cluster it must be annotated with the DNS class `garden` and an annotation denoting the desired DNS names.
 
 Example for an annotated Ingress resource:
 
 ```yaml
-apiVersion: networking.k8s.io/v1
+apiVersion: networking.k8s.io/v1beta1
```
Comment (MartinWeindel) [inline docs/usage/dns_names.md:62]: Please don't use outdated api versions
```suggestion
apiVersion: networking.k8s.io/v1
```

File: docs/usage/dns_names.md [hunk 3]
```diff
@@ -25,95 +25,121 @@ spec:
...
-  tls:
-    - hosts:
-        - test.ingress.my-dns-domain.com
-      secretName: my-cert-secret-name
+  - host: special.example.com
+    http:
+      paths:
+      - backend:
+        serviceName: amazing-svc
+        servicePort: 8080
```
Comment (MartinWeindel) [inline docs/usage/dns_names.md:78]: Please use spec according to `apiVersion: networking.k8s.io/v1`

File: docs/usage/dns_names.md [hunk 4]
```diff
@@ -25,95 +25,121 @@ spec:
...
-              service:
-                name: my-service
-                port:
-                  number: 9000
-            path: /
-            pathType: Prefix
-  tls:
-    - hosts:
-        - test.ingress.my-dns-domain.com
-      secretName: my-cert-secret-name
```
Comment (MartinWeindel) [inline docs/usage/dns_names.md:66]: would be nice to keep the tls section and the commented annotation `#cert.gardener.cloud/purpose: managed` to have a complete example as most users will use both shoot-dns-service and shoot-cert-service

File: docs/usage/dns_names.md [hunk 5]
```diff
@@ -25,95 +25,121 @@ spec:
...
   type: LoadBalancer
 ```
 
-The *dnsnames* annotation accepts a comma-separated list of DNS names, if
-multiple names are required.
-
-For an Ingress, the DNS names are already declared in the specification.
-Nevertheless the *dnsnames* annotation must be present. Here a subset of the 
-DNS names of the ingress can be specified. If DNS names for all names are
-desired, the value `all` can be used.
```
Comment (MartinWeindel) [inline docs/usage/dns_names.md:94]: Why should this useful information be dropped?

File: docs/usage/dns_names.md [hunk 6]
```diff
@@ -25,95 +25,121 @@ spec:
...
+  ttl: 600
+  targets:
+  - 1.2.3.4
+```
 
-If one of the accepted DNS names is a direct subname of the shoot's ingress
-domain, this is already handled by the standard wildcard entry for the ingress
-domain. Therefore this name should be excluded from the *dnsnames* list in the
-annotation. If only this DNS name is configured in the ingress, no explicit 
-DNS entry is required, and the DNS annotations should be omitted at all.
```
Comment (MartinWeindel) [inline docs/usage/dns_names.md:100]: Why should this useful information be dropped?

File: docs/usage/dns_names.md [hunk 7]
```diff
@@ -25,95 +25,121 @@ spec:
...
+You can check the status of the `DNSentry` with
```
Comment (MartinWeindel) [inline docs/usage/dns_names.md:124]: typo

```suggestion
You can check the status of the `DNSEntry` with
```

**Review by MartinWeindel — CHANGES_REQUESTED** [review]
Thanks for the updated documentation.
I have a few minor change requests.

File: docs/usage/dns_names.md (hunk 6 above)
Comment (etiennnr) [reply docs/usage/dns_names.md:100]: You're right, probably deleted that by error

File: docs/usage/dns_names.md (hunk 3 above)
Comment (etiennnr) [reply docs/usage/dns_names.md]: done

**Review by MartinWeindel — APPROVED** [review]
/lgtm
Thanks for the improvements!

**Pull request merged**